sample-docker-app/index.ts

```diff
@@ -1,41 +1,57 @@
-import { Pond, Semantics, OnStateChange, Subscription, FishTypeImpl } from '@actyx/pond'
+import { Pond, Fish, FishId, Tag } from '@actyx/pond'
 
 // Each fish keeps some local state it remembers from all the events it has seen
 type State = { time: string, name: string, msg: string, } | undefined
+const nameTag = Tag<string>('name')
 
-const ForgetfulChatFish: FishTypeImpl<State, string, string, State> = FishTypeImpl.of({
+const mkForgetfulChatFish = (name: string): Fish<State, string> => ({
     // The kind of fish is identified by the meaning of its event stream, the semantics
-    semantics: Semantics.of('ForgetfulChatFish'),
+    fishId: FishId.of('ForgetfulChatFish', name, 0),
 
     // When the fish first wakes up, it computes its initial state and event subscriptions
-    initialState: (_name, _sourceId) => ({
-        state: undefined, // start without information about previous event
-        subscriptions: [Subscription.of(ForgetfulChatFish)] // subscribe across all names
-    }),
+    initialState: undefined, // start without information about previous event
 
     // Upon each new event, keep some details of that event in the state
-    onEvent: (_state, event) => ({
-        time: new Date(event.timestamp / 1000).toISOString(),
-        name: event.source.name,
-        msg: event.payload
-    }),
+    onEvent: (_state, event, metadata) =>
+        ({
+            time: metadata.timestampAsDate().toISOString(),
+            name: metadata.tags.find(x => x.startsWith(
+                `${nameTag.rawTag}:`
+            ))?.split(':')[1] || ":-(",
+            msg: event
+        }),
+    where: nameTag
+});
 
-    // Show the state computed above to the outside world (see Pond.observe below)
-    onStateChange: OnStateChange.publishPrivateState(),
+(async () => {
+    // get started with a Pond
+    const pond = await Pond.default()
+    // figure out the name of the fish we want to wake up
+    const myName = process.argv[2] || pond.info().sourceId
+    // wake up fish of kind ForgetfulChatFish with name myName and log its published states
+    const list = document.getElementById('list')
+    pond.observe(mkForgetfulChatFish(myName), state => {
+        const p = document.createElement('li')
+        p.textContent = JSON.stringify(state)
+        list.appendChild(p)
```

Comment:
I do not like the idea, that we build up a list and when you restart it, it is empty. 

It's very contrary to everything is there for ever